upgrades/upgradevalidation/version_test.go

```diff
@@ -81,55 +81,3 @@ func (s *versionSuite) assertUpgradeControllerAllowed(c *gc.C, i int, t versionC
 		c.Check(err, gc.ErrorMatches, t.err)
 	}
 }
-
-func (s *versionSuite) TestMigrateToAllowed(c *gc.C) {
```

Comment:
this definitely feels like a test we should make sure to keep some cross version compatibility checking that it does allow you to do so and doesn't reject you.